apiary.apib

```diff
@@ -73,17 +73,47 @@ If you find any bug in our API please use [admin@zonky.cz](mailto:admin@zonky.cz
 
 ## Access token and refresh token
-To keep things as secure as possible you have to pay attention to the right usage of OAuth authentication protocol. The right way is to obtain 
+To keep things as secure as possible you have to pay attention to the right usage of OAuth authentication protocol. The right way is to obtain
 an access token and a refresh token with user's credentials *on the first authentication request*. Retrieved access token has limited validity (advertised in the
 response). After its expiration you should retrieve a new one with a refresh token which has longer validity.
 
+## Authorization code grant Flow
+### Client onboarding
+Client application must obtain client credentials before it can use authorization code flow. To start onboarding process please send us email to oauth@zonky.cz
+with following details
+- application name
+- application description
+- requested scopes (list of scopes this app will need to operate)
+- redirect url ( target url where is user redirected after granting authorization code or in case of error) Url is verified for match except for params
+
+Zonky will evaluate your request and if everything is fine we will provide unique clientId and clientSecret for you. **ClientSecret is confidential** and should not
+be shared anywhere. Client id and client secret should be used in communication only between secured servers.
+
+### Authorization flow
+When client need to connect to protected resource and hasn't valid access token it should redirect user to `oauth/athorize` endpoint( find details at endpoint description).
+If user is logged in and client app is not approved zonky will display consent page with some basic info about client app. If client application is already approved
+consent page is not displayed again and fow continues directly. After successful approval user is forwarded to redirect url with authorization code.
+
+Client app can use authorization code to retrieve access token by calling `oauth/token` endpoint. Authorization code can be used only once and expires after 5 min.
+After successful authorization client app can use access token and refresh token for calling secured api. When refresh token expires client app must initiate
+new authorization code flow to obtain access and refresh token.
+
+User can revoke application approval anytime and it will lead to immediate revocation of all tokens.
+
+### Scopes
+Currently we have only one scope, that will give client application all rights to access zonky. It will likely change in future and more detailed scopes will exists.
+| Scope name     | Description           |
+| :------------- |:-------------         |
+| SCOPE_APP_WEB  | Allow all actions with account |
```

Comment:
Last time I tried downloading XLS exports from Zonky, I needed a different scope - SCOPE_APP_WEB didn't work. Don't remember the scope name anymore, just pointing out that this information probably isn't complete.

Reply:
Thanks, you are right there is one more scope. Fixed